---
title: "GxE Multi-Ancestry Supplementary Analysis"
output: workflowr::wflow_html
---

```{r 0-setup, message=FALSE, warning=FALSE, echo=FALSE}
# source('analysis/default/setup.R', local = knitr::knit_global())
```

```{r functions}
library(ggcorrplot)
library(data.table)

options(scipen=999)

# kable function for model comparisons
snipper <- function(tabble){
  # select columns of interest
  micols <- c('comp', 'model', 'mean')
  # subset table
  subtab <- tabble[,micols, with=0]
  # order by component then model)
  subtab <- subtab[order(comp, model),]
  # clean column names
  names(subtab) <- c('Component', 'Model', 'PoVE')
  # simplify PoVE for legibility
  subtab$PoVE <- round(subtab$PoVE, 2)
  #print table
  kable(subtab, 'simple')
}

# plotting function
matter <- function(datanow, fulltitle, montag=NULL, mod7=TRUE){
  plothole <- ggplot(data=datanow, aes(x=fax, y=mean))+
    # scale_color_manual(discrete=T, values = c('A_X1', 'B_X2', 'C_E', 'D_G', 'E_M7', 'F_GxE'), labels = c('X1', 'X2', 'E', 'G', 'M7', 'GxE'))+
    {if(mod7)  scale_color_discrete(labels = c('X1', 'X2', 'E', 'G', 'M7', 'GxE'))}+
    {if(!mod7) scale_color_discrete(labels = c('X1', 'X2', 'E', 'G', 'GxE'))}+
    geom_point(aes(color=comp2))+
    # geom_text(aes(label=modnum))+
    geom_errorbar(aes(ymin = lower, ymax = upper, color=comp2),
                  width = 0.8, linewidth = 0.4, alpha = 1) +
    # labs(title='Model 8 Variance Components')+
    labs(title=fulltitle, color='Component', tag=montag)+
    xlab('Model')+
    # xlab(NULL)+
    ylab('PoVE')+
    theme_minimal()+
    theme(axis.text.x = element_blank(),
          panel.grid.major.x = element_blank(),
          panel.grid.minor.x = element_blank())+
    geom_vline(xintercept = verty, linetype=2, color='lightgray')+
    geom_text(data=modcode, aes(label=mn, x = x, y = y), inherit.aes=FALSE)
  
  return(plothole)
}

# full table of variance component effect sizes
matte <- fread('output/vault/fullport/varcomps.txt')

```

```{r bulk_code}

# Variance component readin ----

# split by trait
mat_dp <- matte[trait=='DP',]
mat_sp <- matte[trait=='SP',]
mat_pp <- matte[trait=='PP',]

# alt model 6 gen pcs
mod1 <- c('6', '6a')
mat_dp_pcs <- mat_dp[modnum %in% mod1,]
mat_sp_pcs <- mat_sp[modnum %in% mod1,]
mat_pp_pcs <- mat_pp[modnum %in% mod1,]

# alt model 6 ancestry grouping
mod2 <- c('6', '6b', '6w')
mat_dp_alt <- mat_dp[modnum %in% mod2,]
mat_sp_alt <- mat_sp[modnum %in% mod2,]
mat_pp_alt <- mat_pp[modnum %in% mod2,]

# base model 6 vs new models 7-9
mod3 <- as.character(6:9)
mat_dp_num <- mat_dp[modnum %in% mod3,]
mat_sp_num <- mat_sp[modnum %in% mod3,]
mat_pp_num <- mat_pp[modnum %in% mod3,]

# Variance component plot formatting ----
# indexes of line breaks - model 6 pcs
# verty <- c(1.5, 3.5, 6.5, 9.5, 13.5, 17.5, 22.5, 27.5, 33.5)
verty <- 0.5 + cumsum(unlist(table(mat_dp_pcs$model)))
verty <- verty[-length(verty)]

# add on hardpoint ends
verty2 <- c(0, verty, nrow(mat_dp_pcs)+1)

# calculates midpoint of each section to place Model Number below x axis
verty3 <- (verty2[2:length(verty2)] - verty2[1:(length(verty2) - 1)]) / 2 + verty2[1:(length(verty2) - 1)]

altnumber <- c('6 - Original', '6a - Alt PCs')

#matter function plots model number at specified breakpoint at y = -1 for each label.
modcode <- data.table(mn=altnumber, x=verty3, y=-1)

dpc <- matter(mat_dp_pcs, fulltitle=NULL, montag='DP', mod7=FALSE)
spc <- matter(mat_sp_pcs, fulltitle=NULL, montag='SP', mod7=FALSE)
ppc <- matter(mat_pp_pcs, fulltitle=NULL, montag='PP', mod7=FALSE)

# print(plot_grid(dpc, spc, ppc, ncol=1))

# indexes of line breaks - model 6 white groups
# verty <- c(1.5, 3.5, 6.5, 9.5, 13.5, 17.5, 22.5, 27.5, 33.5)
verty <- 0.5 + cumsum(unlist(table(mat_dp_alt$model)))
verty <- verty[-length(verty)]

# add on hardpoint ends
verty2 <- c(0, verty, nrow(mat_dp_alt)+1)

# calculates midpoint of each section to place Model Number below x axis
verty3 <- (verty2[2:length(verty2)] - verty2[1:(length(verty2) - 1)]) / 2 + verty2[1:(length(verty2) - 1)]

altnumber <- c('6 - Original', '6b - Brits', '6w - Whites')

#matter function plots model number at specified breakpoint at y = -1 for each label.
modcode <- data.table(mn=altnumber, x=verty3, y=-1)

dalt <- matter(mat_dp_alt, fulltitle=NULL, montag='DP', mod7=FALSE)
salt <- matter(mat_sp_alt, fulltitle=NULL, montag='SP', mod7=FALSE)
palt <- matter(mat_pp_alt, fulltitle=NULL, montag='PP', mod7=FALSE)

# print(plot_grid(dalt, salt, palt, ncol=1))


# indexes of line breaks - models 7-9
# verty <- c(1.5, 3.5, 6.5, 9.5, 13.5, 17.5, 22.5, 27.5, 33.5)
verty <- 0.5 + cumsum(unlist(table(mat_dp_num$model)))
verty <- verty[-length(verty)]

# add on hardpoint ends
verty2 <- c(0, verty, nrow(mat_dp_num)+1)

# calculates midpoint of each section to place Model Number below x axis
verty3 <- (verty2[2:length(verty2)] - verty2[1:(length(verty2) - 1)]) / 2 + verty2[1:(length(verty2) - 1)]

# altnumber <- c('6 - Original', '7 - ','8', '9')
altnumber <- as.character(6:9)

#matter function plots model number at specified breakpoint at y = -1 for each label.
modcode <- data.table(mn=altnumber, x=verty3, y=-1)

dnum <- matter(mat_dp_num, fulltitle=NULL, montag='DP', mod7=TRUE)
snum <- matter(mat_sp_num, fulltitle=NULL, montag='SP', mod7=TRUE)
pnum <- matter(mat_pp_num, fulltitle=NULL, montag='PP', mod7=TRUE)

# print(plot_grid(dnum, snum, pnum, ncol=1))

# ANOVA readin ----

library(ggcorrplot)

# read in data, transpose
temp_alt <- readRDS('output/vault/fullport/anova.rds')
tempt <- t(temp_alt)

```

# Overview

Six subsections:

- New Models
- Alternate Model 6 Populations
- ANOVA: Environment ~ Ancestry
- GxE-GWAS Variant Selection
- GxE-GWAS Variant Analysis
- GxE-GWAS BMI Covariate

## New Models

Models 7, 8, and 9 are new. Model 6 is included for reference to existing terms.

### Formulas

- $\text{Model 6: } \mathbf{y} = \mathbf{X}_{1}\boldsymbol{\beta}_{1} + \mathbf{X}_{2}\boldsymbol{\beta}_{2} + \mathbf{g} + \mathbf{e} + \mathbf{g \times e} + \boldsymbol{\varepsilon}$
- $\text{Model 7: } \mathbf{y} = \mathbf{X}_{1}\boldsymbol{\beta}_{1} + \mathbf{X}_{2}\boldsymbol{\beta}_{2} + \mathbf{g} + \mathbf{e} + \mathbf{i} + \mathbf{g \times e} + \boldsymbol{\varepsilon}$
- $\text{Model 8: } \mathbf{y} = \mathbf{X}_{1}\boldsymbol{\beta}_{1} + \mathbf{X}_{3}\boldsymbol{\beta}_{3} + \mathbf{g} + \mathbf{e} + \mathbf{g \times e} + \boldsymbol{\varepsilon}$
- $\text{Model 9: } \mathbf{y} = \mathbf{X}_{1}\boldsymbol{\beta}_{1} + \mathbf{X}_{4}\boldsymbol{\beta}_{4} + \mathbf{g} + \mathbf{e} + \mathbf{g \times e} + \boldsymbol{\varepsilon}$

### Shared Terms

- $\mathbf{y}$ is a $n$-vector of phenotypic values,
- $\mathbf{X}_{1}$ is a $n \times a$ design matrix of $a=3$ demographic variables (sex, age, age$^2$),
- $\boldsymbol{\beta}_{1}$ is an $a$-vector of their fixed effects,
- $\mathbf{X}_{2}$ is a $n \times b$ design matrix with $b = 10$ columns including the top 10 genetic principal components,
- $\boldsymbol{\beta}_{2}$ is a $b$-vector of their fixed effects,
- $\mathbf{g}$ is a $n$-vector of random additive genetic values, $\mathbf{g} \sim \mathcal{N}(\mathbf{0}, \mathbf{K}_{g.pcr} \sigma^2_{g.pcr})$,
- $\mathbf{e}$ is a $n$-vector of random lifestyle values, $\mathbf{e} \sim \mathcal{N}(\mathbf{0}, \mathbf{K}_e \sigma^2_e)$,
- $\mathbf{g \times e}$ is a $n$-vector of random interaction values, $\mathbf{g \times e} \sim \mathcal{N}(\mathbf{0}, \mathbf{K}_{g.pcr \times e} \sigma^2_{g.pcr \times e})$,
- $\boldsymbol{\varepsilon}$ is a $n$-vector of random residual values, $\boldsymbol{\varepsilon} \sim \mathcal{N}(\mathbf{0}, \mathbf{I}_n \sigma^2_\varepsilon)$.

### Unique Terms

- Model 7: $\mathbf{i}$ is a $n$-vector of random interaction values, $\mathbf{i} \sim \mathcal{N}(\mathbf{0}, \mathbf{K}_{top.plink \times e} \sigma^2_{top.plink \times e})$,
- Model 8: $\mathbf{X}_{3}$ is a $n \times c$ design matrix with $c = 20$ columns including the top 10 genetic principal components (plink), the top 10 eigenvectors of the plink interaction relationship matrix (IRM),
- Model 8: $\boldsymbol{\beta}_{3}$ is a $c$-vector of their fixed effects,
- Model 9: $\mathbf{X}_{4}$ is a $n \times d$ design matrix with $d = 23$ columns including the top 10 genetic principal components, the top 10 eigenvectors of the plink IRM, and the top 3 eigenvectors of the Environmental Relationship Matrix (ERM),
- Model 9: $\boldsymbol{\beta}_{4}$ is a $d$-vector of their fixed effects,

### Descriptions

Model 7 includes an additional variance component $\mathbf{i}$ that implements a modified interaction relationship matrix. The modified IRM is given by $\mathbf{K}_{top.plink \times e} = \mathbf{K}_{top.plink} \odot \mathbf{K}_e$ where $\mathbf{K}_{top.plink}$ is the GRM derived from the top 10 eigenvectors of the full plink GRM.

Model 8 includes the top 10 eigenvectors of the full plink IRM as additional covariates (along with top 10 genetic PCs) for a total of 20 covariates in $\mathbf{X}_3$. This IRM is the Hadamard product of the plink GRM and the full ERM, given by $\mathbf{K}_{g.plink \times e} = \mathbf{K}_{g.plink} \odot \mathbf{K}_e$.

Model 9 expands upon Model 8 by including the top 3 eigenvectors of the full ERM as additional covariates for a total of 23 covariates in $\mathbf{X}_4$.


### Estimates of effect sizes

#### Combined

```{r mult_all, echo=FALSE}

plot_grid(dnum, snum, pnum, ncol=1)

```

#### Diastolic Pressure

```{r mult_dp, echo=FALSE}

plot_grid(dnum, ncol=1)
snipper(mat_dp_num)

```

#### Systolic Pressure

```{r mult_sp, echo=FALSE}

plot_grid(snum, ncol=1)
snipper(mat_sp_num)

```

#### Pulse Pressure

```{r mult_pp, echo=FALSE}

plot_grid(pnum, ncol=1)
snipper(mat_pp_num)

```


## Alternate Model 6 Populations

The model 6 framework is used for the following analyses:

$$\text{Model 6: } \mathbf{y} = \mathbf{X}_{1}\boldsymbol{\beta}_{1} + \mathbf{X}_{2}\boldsymbol{\beta}_{2} + \mathbf{g} + \mathbf{e} + \mathbf{g \times e} + \boldsymbol{\varepsilon}$$

### Genetic Principal Components

In the original model, $\mathbf{X}_2$ is comprised of the top 10 genetic principal components computed by us. In the alternate model, we instead used the top 10 genetic principal components provided by the UK Biobank. The estimation of variance components is shown here:

#### Combined

```{r pcs_all, echo=FALSE}

plot_grid(dpc, spc, ppc, ncol=1)

```

#### Diastolic Pressure

```{r pcs_dp, echo=FALSE}

plot_grid(dpc, ncol=1)
snipper(mat_dp_pcs)

```

#### Systolic Pressure

```{r pcs_sp, echo=FALSE}

plot_grid(spc, ncol=1)
snipper(mat_sp_pcs)

```

#### Pulse Pressure

```{r pcs_pp, echo=FALSE}

plot_grid(ppc, ncol=1)
snipper(mat_pp_pcs)

```

Estimates of effect sizes are consistent between models, suggesting our calculation of genetic PCs is appropriate.


### Ancestry Composition

We assessed the estimate of GxE effect sizes by comparing the multi-ancestry estimates to models using different populations.

In these models, the $\mathbf{g}$ and $\mathbf{g \times e}$ components depend on the following populations:

- Original: 25k individuals multi-ancestry, including 15k whites (British and others)
- Brits: 25k individuals of British ancestry exclusively
- Whites: 25k individuals of white ancestry, including British and others

These results are shown here:

#### Combined
```{r alt_all, echo=FALSE}

plot_grid(dalt, salt, palt, ncol=1)

```

#### Diastolic Pressure

```{r alt_dp, echo=FALSE}

plot_grid(dalt, ncol=1)
snipper(mat_dp_alt)

```

#### Systolic Pressure

```{r alt_sp, echo=FALSE}

plot_grid(salt, ncol=1)
snipper(mat_sp_alt)

```

#### Pulse Pressure

```{r alt_pp, echo=FALSE}

plot_grid(palt, ncol=1)
snipper(mat_pp_alt)

```

For all traits, the PoVE of the genetic component $\mathbf{g}$ increases from the original population to both subsets of white individuals for all three blood pressure traits.

The PoVE for the interaction term $\mathbf{g \times e}$ is not consistent across traits:

- Diastolic pressure : Relative **decrease** of 23-43% (7.37 -> 4.21-5.70)
- Systolic pressure : Relative **decrease** of 15-26% (4.20 - 3.12-3.58)
- Pulse pressure : Relative **increase** of 18-27% (3.05 -> 3.60-3.89)

## ANOVA: Environment ~ Ancestry

Assess environmental effects by ancestry. Analysis of variance models where each environment was regressed across ancestry groups given by $env \sim anc$ where $env$ is a single environment and $anc$ is a factor coding for ancestry, containing the five classes:

- Asian
- Black
- Chinese
- Mixed
- White

### ANOVA Test Results {.tabset}

Ethnicity was extremely significant (p-value less than 1e-300) in the top six environments (alcohol intake, beef, cheese, coffee, pork, and Townsend index) for the respective ANOVA tests. The only environment that did not vary significantly across ancestry groups was daily activity levels (act0_d p = 0.141). All other environments had a significant difference in variance for at least one ancestry group.

#### P-values only

```{r anova_penv}

pvals <- fread('output/anova/misc/pvals.txt')

pvals$`P value` <- scientific(pvals$`P value`)

kable(pvals, 'simple', digits = 4)

```

#### Full Table

```{r anova_pall}

options(scipen = 2)
anovatable <- fread('output/anova/misc/all_anova.txt')
anovatable$Term <- gsub('Ethincity','Ethnicity', anovatable$Term)

anovatable$`Pr(>F)` <- scientific(anovatable$`Pr(>F)`)

kable(anovatable, 'simple', digits=2)

```

#### All Ethnicity Terms

```{r anova_pethn}

anova_ethn <- anovatable[Term=='Ethnicity',]
kable(anova_ethn, 'simple', digits=2)

```

#### All Residual Terms

```{r anova_resids}

anova_resid <- anovatable[Term=='Residuals',]
anova_resid$`F value` <- NULL
anova_resid$`Pr(>F)` <- NULL

kable(anova_resid, 'simple', digits=2)

```

### {-}

### LM Coefficients {.tabset}

#### Clean

```{r anova, echo=FALSE}

ggcorrplot(tempt, lab=0)

```

#### Table

```{r anovatable, echo=FALSE}

kable(temp_alt, 'simple', digits=3)

```

#### Labeled (WIP)

```{r anova_lab, echo=FALSE}

ggcorrplot(tempt, lab=1)

```

#### Labeled + 90 (WIP)

```{r anova_lab90, echo=FALSE}

ggcorrplot(temp_alt, lab=1)

```

### {-}

## GxE-GWAS Variant Selection

The full breakdown of selected SNPs is available in the [QQ Plot Selection report](qqtop.html). These results are summarized here:

```{r qqsnps, echo=FALSE}

qqsnps <- fread('output/vault/fullport/qqsnps.txt')

kable(qqsnps,'simple')

```

## GxE-GWAS Variant Analysis

### Variant Effect Predictor

```{r var_vepall, echo=FALSE}

vepall <- fread('output/vault/fullport/vep.txt')

kable(vepall, 'simple')

```

Ensembl's variant effect predictor (VEP) tool identified 30, 89, and 104 genes for diastolic, systolic, and pulse pressure respectively.

Some of these genes were identified in more than one trait and have been included below:

```{r var_vepjoin, echo=FALSE}

vepjoin <- fread('output/vault/fullport/vep_join.txt')

kable(vepjoin, 'simple')

```

### g:Profiler Terms

[g:Profiler](https://biit.cs.ut.ee/gprofiler/gost): a functional analysis tool provided by the University of Tartu.

Piping the identified genes into g:Profiler yields significant GO terms associated with the provided pool of genes. Results are shown here:

```{r var_gprof, echo=FALSE}

profdat <- fread('output/vault/fullport/final_gprof.txt')

kable(profdat, 'simple')

```

Functional analysis did not identify any significant GO terms, KEGG terms, or CORUM terms for diastolic pressure.

An in depth breakdown of significant variants can be found in the rewrite of the results discussion under GxE-variance-ukb-multi-manuscript/WIP/gxegwas.tex through Overleaf (not linked for permissions and safety reasons).

## GxE-GWAS BMI Covariate {.tabset}

We also tested the inclusion of BMI as an additional covariate. The full comparison of QQ plots is available in the [BMI Covariate QQ Comparison report](bmi_comp.html). 

### Summary

Average Pearson and Spearman correlation coefficients across all environments for each trait.

```{r bmicov}

bmidata <-  fread('output/vault/fullport/bmicov.txt')

pcol <- bmidata[,mean(pearson), by=trait]
scol <- bmidata[,mean(spearman), by=trait]

bmicol <- cbind(pcol, scol)

names(bmicol) <- c('Trait', 'Pearson', 'remove', 'Spearman')

bmicol$remove <- NULL

kable(bmicol, 'simple', digits = 5)

```


### Full Table

Pearson and Spearman correlation coefficients for all trait and environment combinations.

```{r bmifull}

names(bmidata) <- c('Trait', 'Environment', 'Pearson', 'Spearman')

kable(bmidata, 'simple', digits = 5)

```


## {-}

Results are highly correlated, suggesting the inclusion of BMI as a covariate does not have a dramatic effect on results. Lowest correlation is for diastolic pressure at 0.83
